happytransformer/happy_generation.py

```diff
@@ -0,0 +1,160 @@
+
+        model = AutoModelForCausalLM.from_pretrained(model_name)
+
+        super().__init__(model_type, model_name, model)
+
+        self._trainer = GENTrainer(self.model, model_type, self.tokenizer, self._device, self.logger)
+
+    def __assert_default_text_is_val(self, text):
+
+        if not isinstance(text, str):
+            raise ValueError("The text input must be a string")
+        if not text:
+            raise ValueError("The text input must have at least one character")
+
+
+    def generate_text(self, text, settings=GEN_BEAM_SETTINGS,
+                      min_length=20, max_length=60) -> GenerationResult:
+        """
+        :param text: starting text that the model uses to generate text with.
+        :param settings: A dictionary that contains settings that determine what
+         algorithm is used to generate text
+        :param min_length: The minimum number of tokens for the output
+        :param max_length: The maximum number of tokens for the output
+        :return: Text that the model generates.
+        """
+
+        self.__assert_default_text_is_val(text)
+        self.__validate_settings(settings, min_length, max_length)
+
+        input_ids = self.tokenizer.encode(text, return_tensors="pt")
+        adjusted_min_length = min_length + len(input_ids[0])
+        adjusted_max_length = max_length + len(input_ids[0])
+
+        output = self.model.generate(input_ids,
+                                     **settings,
+                                     min_length=adjusted_min_length,
+                                     max_length=adjusted_max_length,
+                                     )
+        result = self.tokenizer.decode(output[0], skip_special_tokens=True)
+        final_result = self.__gt_post_processing(result, text)
+
+        return GenerationResult(text=final_result)
+
+
+
+    def __gt_post_processing(self, result, text):
+        """
+        A method for processing the output of the model. More features will be added later.
+        :param result: result the output of the model after being decoded
+        :param text:  the original input to generate_text
+        :return: returns to text after going through post-processing. Removes starting text
+        """
+
+        return result[len(text):]
+
+    def __validate_settings(self, settings, min_length, max_length):
+        """
+        The default function for min_length and max_length within Hugging Face is for the
+        min/max  number of tokens of the combined input/output .
+
+        We believe it makes more sense for them to be just for the output.
+
+        So, we added custom min_length and max_length to the generate function.
+
+        We must ensure that the user does not include
+        min_length/max_length within the input dictionary
+        """
+        if "min_length" in settings:
+            self.logger.warning("\"min_length\" is a parameter for the method \"generate_text\"."
```

Comment:
backslashes can be avoided by using single quotes for the outermost

```
print('look i have "quotes" in my string')
```

Reply:
55438b5

Reply:
c30e234